contracts/Comptroller.sol

```diff
@@ -1016,6 +1019,37 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
         allMarkets.push(vToken);
     }
 
+    /**
+      * @notice delete the market to the markets mapping and set it as unlisted
+      * @dev Admin function to set isListed to false and delete support for the market
+      * @param vToken The address of the market (token) to list
+      * @return uint 0=success, otherwise a failure. (See enum Error for details)
+      */
+    function _unlistMarket(VToken vToken) external returns (uint) {
+        ensureAllowed("_unlistMarket(address)");
+
+        if (!markets[address(vToken)].isListed) {
+            return fail(Error.MARKET_ALREADY_UNLISTED, FailureInfo.SUPPORT_MARKET_DOESNOT_EXISTS);
+        }
+
+        vToken.isVToken(); // Sanity check to make sure its really a VToken
+
+        _addMarketInternal(vToken);
```

Comment:
```suggestion
        _removeMarketInternal(vToken);
```

Reply:
@KJhellico thanks updated the commit. fixed